### PR TITLE
Subtyping Bug

### PR DESCRIPTION
**Pull request by albsch**
Fixed a rare edge case where the BDD representation for ANY or EMPTY did not correspond to a ANY or EMPTY representation of a tuple or function. Added test cases.

No performance impact, unit tests are a few seconds faster in total.

Based on #76, merge that first.

### CODE REVIEW

File: src/erlang_types/dnf_ty_tuple.erl [hunk 1]
```diff
@@ -28,15 +28,17 @@ is_empty(TyBDD) ->
   gen_bdd:dnf(?P, TyBDD, {fun is_empty_coclause/3, fun gen_bdd:is_empty_union/2}).
 
 is_empty_coclause(Pos, Neg, T) ->
-  case bdd_bool:empty() of
-    T -> true;
-    _ -> case Pos of
-           % TODO check correctness of this
```
Comment (albsch) [inline src/erlang_types/dnf_ty_tuple.erl:34]: This was the reason for the bug.

**Review by skogsbaer — APPROVED** [review]
